Updated AssetList Component w/ mock data

joeadams04 opened the pull request:
Removed 'Edited recently'
Updated AssetList Container
New AssetItem Component
Layout fixes

Currently using mock data to populate table

rjw57 commented on src/containers/AssetList.js:
```diff
@@ -1,62 +1,116 @@
-          </TableRow>
-        </TableBody>
-      </Table>
+  	<div className="App-main">
```
Can we use an indent of 2 spaces here to be consistent with the rest of the code?

rjw57 commented on src/style/App.css:
```diff
@@ -39,3 +47,8 @@
+.Asset-item {
+  display: block;
+  width: 100%;
+}
```
Can we move this to components/AssetListItem.css and import it from AssetListItem.js? Also, I don't see where this class is actually applied to an element. Is it implicit?

rjw57 commented:
The [test suite output](https://travis-ci.org/uisautomation/iar-frontend/builds/333154920) includes a lot of warnings. Can these be addressed in this PR?

```
 PASS  src/containers/AssetForm.test.js (5.425s)
 PASS  src/containers/AssetList.test.js (5.804s)
  ● Console
    console.error node_modules/fbjs/lib/warning.js:33
      Warning: Each child in an array or iterator should have a unique "key" prop.
      
      Check the render method of `AssetList`. See https://fb.me/react-warning-keys for more information.
          in AssetListItem (at AssetList.js:100)
          in AssetList (at AssetList.test.js:8)
          in MuiThemeProvider (at testutils.js:13)
          in Router (created by MemoryRouter)
          in MemoryRouter (at testutils.js:13)
```

rjw57 commented on src/containers/AssetList.js:
```diff
@@ -65,52 +65,50 @@ const assetData = [
-  '/assets/dept': 'Assets: My department',
-  '/assets/all': 'Assets: All',
+	'/assets/dept': 'Assets: My department',
+	'/assets/all': 'Assets: All',
```
I think the consistent indent size should actually be 2 to match, e.g. Sidebar.js, Static.js, AssetFormHEader.js. Maybe add a [*.js] section to .editorconfig to this effect?

joeadams04 replied:
Spaces or tabs?

joeadams04 replied:
Ignore that last comment, changes my tab size to 2 spaces.

rjw57 replied:
Nooo..... spaces! If you use a tab size of 2, GitHub will still display them as 8.

joeadams04 commented:
Fixed indentation issues throughout AssetList.

Regarding the styling, the class was used when I was playing around with the layout, however in the end wasn't needed. I've removed it now. From now on, yes we should probably create separate style sheets for each component.

rjw57 commented:
Thanks, as you can see, I think the indent should be 2 spaces, not 4. Also, are your changes the source of the warnings in the test suite output?

rjw57 commented on src/containers/AssetList.js:
```diff
@@ -97,7 +97,7 @@ const AssetList = ({ match }) => (
           {assetData.map( (asset, index) => (
             <AssetListItem
-              index={asset.index}
+              key={index}
```
When we have a real API, this should be asset.id or similar. For the moment, this is fine.

rjw57 commented on src/containers/AssetList.js:
```diff
@@ -1,58 +1,110 @@
-import React from 'react';
+import React, {Component} from 'react';
 import { Link } from 'react-router-dom'
```
In testing I get the following:

```
./src/containers/AssetList.js
  Line 1:  'Component' is defined but never used  no-unused-vars
  Line 2:  'Link' is defined but never used       no-unused-vars
```

It's so trivial that I'll just update the PR to remove these and then accept.
